Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -101,29 +102,32 @@ sub _releases {
 			$_->{release_type} = 'COMPILATION';
 			$addToMainReleases->();
 			# only list outside the compilations if Composer/Conductor
-			next unless $_->{role_ids} =~ /COMPOSER|CONDUCTOR/ && $_->{role_ids} !~ /ARTIST|BAND/;
+			next unless ( $defaultRoles =~ /COMPOSER|CONDUCTOR/ || $userDefinedRoles ) && $defaultRoles !~ /ARTIST|BAND/;
 		}
 		# Release Types if album artist
 		elsif ( $_->{role_ids} =~ /ALBUMARTIST/ ) {
-			$addToMainReleases->();
-			next;
+			$addToMainReleases->() if $defaultRoles;
+			next unless $userDefinedRoles;
 		}
 		# Consider this artist the main (album) artist if there's no other, defined album artist
 		elsif ( $_->{role_ids} =~ /ARTIST/ ) {
```

Comment:
Again: is it one to match `TRACKARTIST` here? Should we check word boundaries?

Reply:
This triggered a thought: all the `$addToMainReleases` types should only be default roles, so we should be testing `$defaultRoles`, not `$_->{role_ids}`.

Simplifies things a little and I think makes stricter matching redundant.

See incoming push.